### PR TITLE
Require HiOp 0.7.1 for ExaGO 1.5.0+.

### PR DESCRIPTION
**Pull request by cameronrutherford**
HiOp's develop branch has breaking API changes with header files which requires a patch to ExaGO.

We will limit compatibility up to 0.7.1 until a new HiOp patch and associated ExaGO patch.

### CODE REVIEW

**Review by eugeneswalker — APPROVED** [review]
We can wait for other maintainer to comment or merge whenever you say so.

**Comment by cameronrutherford** [comment]
> We can wait for other maintainer to comment or merge whenever you say so.

This is a small enough change so I think we can just merge.

**Comment by eugeneswalker** [comment]
> > We can wait for other maintainer to comment or merge whenever you say so.
> 
> This is a small enough change so I think we can just merge.

Perfect. I turned auto-merge on and it should merge once CI checks finish.

**Pull request merged**
